File: functions/alsa-lib/variants/c/1_iec958_parity_4.c
#include <stdint.h>

static unsigned int iec958_parity(unsigned int data)
{
	data = data >> 4 & 0x1FFFFFFF; // Masking to 28 bits
	int count = __builtin_popcount(data);
	return count & 1;
}


int main(){
    iec958_parity(0);
}

File: functions/alsa-lib/variants/go/1_iec958_parity_4.go
package main

func iec958_parity(data uint32) uint32 {
	data >>= 4
	parity := uint32(0)
	for bit := 4; bit <= 30; bit++ {
		parity ^= data & 1
		data >>= 1
	}
	return parity
}


func main(){
    iec958_parity(0)
}